src/GafferScene/RenderController.cpp

```diff
@@ -437,6 +437,14 @@ class RenderController::SceneGraph
 
 			clean( TransformComponent );
 
+			// VisibleSet
+
+			if( ( m_dirtyComponents & VisibleSetComponent ) && updateVisibleSet( path, controller->m_visibleSet, controller->m_minimumExpansionDepth ) )
+			{
+				m_changedComponents |= VisibleSetComponent;
+				m_dirtyComponents |= ObjectComponent;
```

Comment:
It seems that this will lead to more `updateObject()` calls than we would have made in the past. I think technically we only need to dirty `ObjectComponent` if the draw mode has changed, but not if the descendant visibility has? Is that a worthwhile optimisation?

Reply:
Yep, good point! I've added this in b2a0dcf. I'd hoped to include a test but I've struggled to demonstrate it with the CapturingRenderer. We were generating more `updateObject()` calls than necessary, but I suspect we were not actually updating the object due to the hashes remaining consistent when getting the object samples [here](https://github.com/GafferHQ/gaffer/blob/3147a1980647e13ce339b08e43eff4f8482e26ea/src/GafferScene/RenderController.cpp#L832-L835)?

If you have a better idea for a test, I'd be all for it.

Reply:
Ah, that would make sense. Perhaps we can use a PerformanceMonitor to check that `objectSamples()` isn't called at all? Something like this?

```
Gaffer.ValuePlug.clearCache()
Gaffer.ValuePlug.clearHashCache()
with Gaffer.PerformanceMontitor() as monitor :
    controller.setVisibleSet( ... )
    controller.update()

self.assertEqual( monitor.plugStatistics( node["out"]["object"] ).hashCount, 0 )
self.assertEqual( monitor.plugStatistics( node["out"]["object"] ).computeCount, 0 )
```

Reply:
Thanks, that works nicely. I've added a test along those lines to 1a9b995 and confirmed that it fails with the optimisation removed.